packages/plugins/connections/connection-redis/src/connections/Redis/Redis/Redis.test.js

```diff
@@ -0,0 +1,57 @@
+/*
+  Copyright 2020-2021 Lowdefy, Inc
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { validate } from '@lowdefy/ajv';
+import Redis from './Redis.js';
+
+const { checkRead, checkWrite } = Redis.meta;
+const schema = Redis.schema;
+
+test('valid request schema, with url', () => {
+  const request = {
+    command: 'set',
+    parameters: ['key', 10],
+  };
+  expect(validate({ schema, data: request })).toEqual({ valid: true });
+});
+
+test('command is not a string', () => {
+  const request = {
+    command: true,
+    parameters: ['key', 'value'],
+  };
+  expect(() => validate({ schema, data: request })).toThrow(
+    'Redis request property "command" should be a string.'
+  );
+});
+
+test('parameters is not an array', () => {
+  const request = {
+    command: 'set',
+    parameters: 'string',
+  };
+  expect(() => validate({ schema, data: request })).toThrow(
+    'Redis request property "parameters" should be an array.'
+  );
+});
+
+test('checkRead should be false', async () => {
+  expect(checkRead).toBe(false);
+});
+
+test('checkWrite should be false', async () => {
+  expect(checkWrite).toBe(false);
+});
```

Comment:
Please add tests with a mocked redis connection to test that the client and command functions receive the correct arguments, all the commands are converted to uppercase, errors like invalid command are thrown correctly etc.

This is a possible example of the mocks should look like, but I have not been able to test/run this so I am not sure that it will work:
```js
jest.mock('redis', () => {
  return {
    createClient: mockCreateClient,
  };
});

const mockCreateClient = jest.fn();

let mockCreateClient;

beforeAll(async () => {
  mockCreateClient = (await import('redis')).createClient;
});

test("mocked client", () => {
  const client = {
    on: jest.fn(),
    connect: jest.fn(),
    GET: jest.fn(),
  }
  mockCreateClient.mockImplementation(() => client);
  expect(mockCreateClient.mock.calls).toEqual(/*...*/);
  expect(client.connect.mock.calls).toEqual(/*...*/);
  expect(client.GET.mock.calls).toEqual(/*...*/);
})
```

Reply:
@SamTolmay The example works mock implementation works fine. 

Though I spent last Friday trying to make it work just to realise jest mocks were not working correctly.
It seems to be an issue with jest and swc, you can see it in [this thread](https://github.com/swc-project/jest/issues/14).

It was fixed with a patch [here](https://github.com/swc-project/jest/pull/44#issuecomment-980601069).

This issue is affecting the tests on the other plugins too, but for now I bumped the @swc/core in the connection-redis only and the mocks work correctly there.

Reply:
Okay thanks. I think we might have been running into this too. We will update the version for the other packages once we have merged this PR.